src/network/network.ts

```diff
@@ -73,12 +86,26 @@ export class Network {
    * @param node - The network node to be removed.
    */
   public removeNode(node: NetworkNode) {
+    if (this.isRemoved()) {
+      throw new GolemNetworkError(
+        `Unable to remove node ${node.id} from removed network`,
+        NetworkErrorCode.NetworkRemoved,
+        this.getNetworkInfo(),
+      );
+    }
     if (!this.hasNode(node)) {
       throw new GolemNetworkError(`There is no node ${node.id} in the network`, NetworkErrorCode.NodeRemovalFailed);
     }
     this.nodes.delete(node.id);
   }
 
+  public remove() {
```

Comment:
`remove` implies that this does something (like call yagna) but it only sets the state as removed. Maybe we could rename it to something like `markAsRemoved`?